packages/core/src/internal/proc.js

```diff
@@ -327,23 +280,18 @@ export default function proc(env, iterator, parentContext, parentEffectId, meta,
     } else if (is.iterator(effect)) {
       resolveIterator(effect, effectId, meta, currCb)
     } else if (effect && effect[IO]) {
-      const { type, payload } = effect
-      if (type === effectTypes.TAKE) runTakeEffect(payload, currCb)
-      else if (type === effectTypes.PUT) runPutEffect(payload, currCb)
-      else if (type === effectTypes.ALL) runAllEffect(payload, effectId, currCb)
-      else if (type === effectTypes.RACE) runRaceEffect(payload, effectId, currCb)
-      else if (type === effectTypes.CALL) runCallEffect(payload, effectId, currCb)
-      else if (type === effectTypes.CPS) runCPSEffect(payload, currCb)
-      else if (type === effectTypes.FORK) runForkEffect(payload, effectId, currCb)
-      else if (type === effectTypes.JOIN) runJoinEffect(payload, currCb)
-      else if (type === effectTypes.CANCEL) runCancelEffect(payload, currCb)
-      else if (type === effectTypes.SELECT) runSelectEffect(payload, currCb)
-      else if (type === effectTypes.ACTION_CHANNEL) runChannelEffect(payload, currCb)
-      else if (type === effectTypes.FLUSH) runFlushEffect(payload, currCb)
-      else if (type === effectTypes.CANCELLED) runCancelledEffect(payload, currCb)
-      else if (type === effectTypes.GET_CONTEXT) runGetContextEffect(payload, currCb)
-      else if (type === effectTypes.SET_CONTEXT) runSetContextEffect(payload, currCb)
-      else currCb(effect)
```

Comment:
so we don't need this logic branch. do we?

Reply:
We do not need these if-else anymore, the branch logic is in this single statement: `const effectRunner = effectRunnerMap[effect.type]`.